tests/lib/rules/no-array-constructor.js

```diff
@@ -432,3 +432,105 @@ ruleTester.run("no-array-constructor", rule, {
         }))
     ]
 });
+
+const ruleTesterTypeScript = new RuleTester({
+    languageOptions: {
+        parser: require("@typescript-eslint/parser")
+    }
+});
+
+ruleTesterTypeScript.run("no-array-constructor", rule, {
+    valid: [
+        "new Array(x);",
+        "Array(x);",
+        "new Array(9);",
+        "Array(9);",
+        "new foo.Array();",
+        "foo.Array();",
+        "new Array.foo();",
+        "Array.foo();",
+
+        // TypeScript
+        "new Array<Foo>(1, 2, 3);",
+        "new Array<Foo>();",
+        "Array<Foo>(1, 2, 3);",
+        "Array<Foo>();"
+
+        /*
+         * optional chain
+         * 'Array?.(x);',
+         * 'Array?.(9);',
```

Comment:
Should we allow code with optional chaining to support TypeScript syntax?

Reply:
Optional chaining isn't a TypeScript syntax. Since [`Array?.()` is already reported by the rule](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXJyYXktY29uc3RydWN0b3I6IFwiZXJyb3JcIiovXG5cbkFycmF5KCk7XG5cbkFycmF5Py4oKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=) I think it should be here too, yeah. Filed an issue on the extension rule: https://github.com/typescript-eslint/typescript-eslint/issues/10933

Reply:
But right now rule just report `Array?.()` not Array?.(1) should we keep this behavior or reporting all the code with optional chaining?

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXJyYXktY29uc3RydWN0b3I6IFwiZXJyb3JcIiovXG5cbkFycmF5KCk7XG5BcnJheSgxKTtcblxuQXJyYXk/LigpO1xuQXJyYXk/LigxKTtcbiIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

Reply:
`Array?.(1)` is not reported because it has exactly one argument. It creates an array of length 1, and not array that contains the number 1 as an element. So `Array?.(1)` cannot be replaced with `[1]`.

For the same reason `Array(foo)` isn't reported, since it has one argument, but `Array(foo, bar)` is.

Reply:
Got it, thanks!